### PR TITLE
ResStock-HPXML

### LINKED ISSUE #725: Update buildstockbatch workflow generator
Moved from: https://trello.com/c/Mniz4s06/10-update-buildstockbatch-workflow-generator.

Checklist:
- [ ] Merge https://github.com/NREL/buildstockbatch/pull/208

### PR DESCRIPTION
**Pull request by joseph-robertson**
## Pull Request Description

Closes https://github.com/NREL/resstock/issues/725.

This is the companion PR to: NREL/resstock#443

- [x] Remove `include_timeseries_unmet_loads` (https://github.com/NREL/OpenStudio-HPXML/pull/846)

## Checklist

Not all may apply

- [ ] Code changes (must work)
- [ ] Tests exercising your feature/bug fix (check coverage report on CircleCI build -> Artifacts)
- [ ] All other unit tests passing
- [ ] Update validation for project config yaml file changes
- [ ] Update existing documentation
- [ ] Run a small batch run to make sure it all works (local is fine, unless an Eagle specific feature)
- [ ] Add to the changelog_dev.rst file and propose migration text in the pull request

### CODE REVIEW

File: buildstockbatch/aws/aws.py [hunk 1]
```diff
@@ -1784,6 +1784,7 @@ def run_batch(self):
                 project_path = pathlib.Path(self.project_dir)
                 buildstock_path = pathlib.Path(self.buildstock_dir)
                 tar_f.add(buildstock_path / 'measures', 'measures')
+                tar_f.add(buildstock_path / 'resources/hpxml-measures', 'resources/hpxml-measures')
```
Comment (nmerket) [inline buildstockbatch/aws/aws.py]: Should probably make this conditional as this directory won't be available for other types of runs.

File: buildstockbatch/base.py [hunk 2]
```diff
@@ -38,8 +38,9 @@
 
 class BuildStockBatchBase(object):
 
-    DEFAULT_OS_VERSION = '2.9.1'
-    DEFAULT_OS_SHA = '3472e8b799'
+    # http://openstudio-builds.s3-website-us-east-1.amazonaws.com
+    DEFAULT_OS_VERSION = '3.1.0'
+    DEFAULT_OS_SHA = 'e165090621'
```
Comment (nmerket) [inline buildstockbatch/base.py]: As we talked about we need to find a way (eventually) to make this dependent on which workflow you're using. Need to coordinate with ComStock.

File: buildstockbatch/eagle.py [hunk 3]
```diff
@@ -229,6 +229,10 @@ def run_job_batch(self, job_array_number):
             pathlib.Path(self.buildstock_dir) / 'measures',
             self.local_buildstock_dir / 'measures'
         )
+        self.clear_and_copy_dir(
+            pathlib.Path(self.buildstock_dir) / 'resources/hpxml-measures',
+            self.local_buildstock_dir / 'resources/hpxml-measures'
+        )
```
Comment (nmerket) [inline buildstockbatch/eagle.py]: Again should probably make conditional for when this doesn't exist.

File: buildstockbatch/workflow_generator/test_workflow_generator.py [hunk 4]
```diff
@@ -286,3 +287,68 @@ def test_simulation_output(mocker):
     args = osw['steps'][-2]['arguments']
     for argname in ('include_enduse_subcategories',):
         assert(args[argname] != default_args[argname])
+
+
+def test_residential_hpxml(mocker):
```
Comment (nmerket) [inline buildstockbatch/workflow_generator/test_workflow_generator.py:320]: 👍

File: docs/workflow_generators/residential_hpxml.rst [hunk 5]
```diff
@@ -0,0 +1,49 @@
+Residential HPXML Workflow Generator
+------------------------------
```
Comment (nmerket) [inline docs/workflow_generators/residential_hpxml.rst]:
```suggestion
Residential HPXML Workflow Generator
------------------------------------
```

**Review by nmerket — COMMENTED** [review]
That was fast. A few comments below.

File: buildstockbatch/localdocker.py [hunk 6]
```diff
@@ -99,6 +99,7 @@ def run_building(cls, project_dir, buildstock_dir, weather_dir, docker_image, re
         bind_mounts = [
             (sim_dir, '', 'rw'),
             (os.path.join(buildstock_dir, 'measures'), 'measures', 'ro'),
+            (os.path.join(buildstock_dir, 'resources', 'hpxml-measures'), 'resources/hpxml-measures', 'ro'),
```
Comment (joseph-robertson) [inline buildstockbatch/localdocker.py]: Make conditional here, too.

File: buildstockbatch/postprocessing.py [hunk 7]
```diff
@@ -107,12 +111,30 @@ def read_out_osw(fs, filename):
...
+        with fs.open(started, 'r') as f:
+            started_at = re.search(r'Started Workflow (.*\s.*?)\s', f.readline()).group(1)
+            started_at = started_at.replace('-', '').replace(':', '').replace(' ', 'T') + 'Z'
+            jobs['started_at'] = started_at
+    except:
+        return None
+    try:
+        with fs.open(finished, 'r') as f:
+            completed_at = re.search(r'Finished Workflow (.*\s.*?)\s', f.readline()).group(1)
+            completed_at = completed_at.replace('-', '').replace(':', '').replace(' ', 'T') + 'Z'
```
Comment (shorowit) [inline buildstockbatch/postprocessing.py]: @joseph-robertson Is there really not a more python-ic way to convert from the one time string to the other?

Comment (shorowit) [reply buildstockbatch/postprocessing.py]: https://docs.python.org/3/library/time.html

Comment (nmerket) [reply buildstockbatch/postprocessing.py]: Or more specifically...

https://docs.python.org/3/library/datetime.html#strftime-and-strptime-behavior

Comment (joseph-robertson) [reply buildstockbatch/postprocessing.py]: 👀

File: buildstockbatch/base.py (hunk 2 above)
Comment (joseph-robertson) [reply buildstockbatch/base.py]: @nmerket Idea: Pull openstudio version from resstock's `__version__.py` file? https://github.com/NREL/resstock/blob/develop/resources/__version__.py#L5

File: buildstockbatch/base.py [hunk 8]
```diff
@@ -39,8 +39,8 @@
 class BuildStockBatchBase(object):
 
     # http://openstudio-builds.s3-website-us-east-1.amazonaws.com
-    DEFAULT_OS_VERSION = 'dev-3.3.0-rc2'
-    DEFAULT_OS_SHA = '45b36b8d4c'
+    DEFAULT_OS_VERSION = '3.3.0'
```
Comment (joseph-robertson) [inline buildstockbatch/base.py:42]: 👍

**Comment by joseph-robertson** [comment]
@nmerket @rajeee Maybe we can incorporate "new tests" on this branch? It's needed for resstock, and current tests are failing anyways.

**Review by nmerket — APPROVED** [review]
I think this looks good. It seems like you've been using it for a while now and it's working. There's some stuff I'd like to generalize more like bind mounts, but I think we can do that with another PR.

**Pull request merged**